module_utils/k8s_common.py

```diff
@@ -197,7 +197,7 @@ def execute_module(self):
         else:
             if not existing:
                 k8s_obj = self._create(namespace)
-                return_attributes[self.kind] = k8s_obj.to_dict()
+                return_attributes[self.kind] = k8s_obj.to_dict() if k8s_obj else k8s_obj
```

Comment:
OK. I get it now. This change makes sense, with one exception. Instead of return `k8s_obj`, let's return dict(). 

Also, can you make the same change on line #165, #170, #213, and #238. Seems like that would catch all cases where this problem might happen.

Thanks!

Reply:
I updated the PR and used `{}` instead of `dict()` since the latter does not seem to be used in the rest of the codebase. If you prefer `dict()` above `{}` let me know and I will update the PR.